Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/worker/RatResultRetrievalWorker.kt

```diff
@@ -0,0 +1,94 @@
+package de.rki.coronawarnapp.coronatest.worker
+
+import android.content.Context
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
+import de.rki.coronawarnapp.coronatest.CoronaTestRepository
+import de.rki.coronawarnapp.coronatest.execution.TestResultScheduler
+import de.rki.coronawarnapp.coronatest.execution.TestResultScheduler.RatPollingMode.DISABLED
+import de.rki.coronawarnapp.coronatest.execution.TestResultScheduler.RatPollingMode.PHASE1
+import de.rki.coronawarnapp.coronatest.execution.TestResultScheduler.RatPollingMode.PHASE2
+import de.rki.coronawarnapp.coronatest.latestRAT
+import de.rki.coronawarnapp.coronatest.type.CoronaTest
+import de.rki.coronawarnapp.util.TimeStamper
+import de.rki.coronawarnapp.util.worker.InjectedWorkerFactory
+import de.rki.coronawarnapp.worker.BackgroundConstants
+import kotlinx.coroutines.flow.first
+import org.joda.time.Duration
+import timber.log.Timber
+
+/**
+ * Diagnosis test result retrieval by periodic polling
+ */
+class RatResultRetrievalWorker @AssistedInject constructor(
+    @Assisted val context: Context,
+    @Assisted workerParams: WorkerParameters,
+    private val coronaTestRepository: CoronaTestRepository,
+    private val timeStamper: TimeStamper,
+    private val testResultScheduler: TestResultScheduler,
+) : CoroutineWorker(context, workerParams) {
+
+    override suspend fun doWork(): Result {
+        Timber.tag(TAG).d("$id: doWork() started. Run attempt: $runAttemptCount")
+
+        if (runAttemptCount > BackgroundConstants.WORKER_RETRY_COUNT_THRESHOLD) {
+            Timber.tag(TAG).d("$id doWork() failed after $runAttemptCount attempts. Rescheduling")
+
+            testResultScheduler.setRatResultPeriodicPollingMode(mode = testResultScheduler.ratResultPeriodicPollingMode)
+            Timber.tag(TAG).d("$id Rescheduled background worker")
+
+            return Result.failure()
+        }
+
+        // checking abort conditions
+        val rat = coronaTestRepository.latestRAT.first()
+        if (rat == null) {
+            Timber.tag(TAG).w("There is no PCR test available!?")
+            stopWorker()
```

Comment:
Can we rename this `disablePolling`?  `stopWorker` sounds like just stopping "this" worker execution.

Reply:
done!